### PR TITLE
automatic sweep registration

### PR DESCRIPTION
**Pull request by hyenal**
Remove the need for manually specifying whether we are using a W&B sweep or not. Also linked to this idea https://github.com/martinsbruveris/tensorflow-image-models/discussions/36

### CODE REVIEW

File: tfimm/train/train.py [hunk 1]
```diff
@@ -80,9 +76,10 @@ def run(cfg: Union[ExperimentConfig, dict], parse_args: bool = True):
             resume=False,
         )
 
-    # In case we are using W&B sweep we need to adjust project name etc. to make them
-    # specific for each run in the sweep.
-    if cfg.sweep:
+    # In case we are using W&B sweep we need to adjust run name and ckpt dir
+    # To make the specific for each run in the sweep.
```
Comment (martinsbruveris) [inline tfimm/train/train.py:80]: Could you add a comment here to the effect of: When using sweeps, `wandb` sets the env variable `WANDB_RUNQUEUE_ITEM_ID`. The presence/absence of this variable allows us to detect if this run is part of a sweep.

And also add the more detailed comments that were previously part of the config class (L39-41) to explain why we are doing this.

**Review by martinsbruveris — COMMENTED** [review]
Looks good!

File: tfimm/train/train.py (hunk 1 above)
Comment (martinsbruveris) [reply tfimm/train/train.py]: Never mind, I added the comments and merged the PR.

**Comment by martinsbruveris** [comment]
Thanks for the PR!

**Comment by hyenal** [comment]
Really sorry I saw your messages too late :/ Glad it's merged though

**Pull request merged**